src/qt/ve_qitems_mqtt.cpp

```diff
@@ -377,8 +399,14 @@ void VeQItemMqttProducer::onSubscriptionMessageReceived(const QMqttMessage &mess
 	const QString notificationPrefix = QStringLiteral("N/%1").arg(mPortalId);
 	if (topicName.startsWith(notificationPrefix)) {
 		const QString keepaliveTopic = notificationPrefix + QStringLiteral("/keepalive");
+		const QString heartbeatTopic = notificationPrefix + QStringLiteral("/heartbeat");
 		if (topicName.compare(keepaliveTopic, Qt::CaseInsensitive) == 0) {
 			// ignore keepalive topic.
+		} else if (topicName.compare(heartbeatTopic, Qt::CaseInsensitive) == 0) {
+			// (re)start our heartbeat timer.
+			mHeartBeatTimer->start();
+			mMissedHeartbeats = 0;
```

Comment:
Is `mMissedHeartbeats` needed? It seems the value is not actually used anywhere.

Reply:
Also see my comment. It can/should be used, to have three heartbeat states.

Reply:
Thanks, updated.

Just double checking: is it correct that it should transition to HeartbeatMissed when count = 2, and to HeartbeatInactive when count = 3?  That is my reading of the doc, but perhaps I misunderstand and it should be HeartbeatMissed if count is 1 or 2, and HeartbeatInactive if count is 3?

Reply:
(Also updated interval to 5s as per doc and your comment - thanks)

Reply:
@chriadam The values 2 and 3 are perhaps too high; they came from the algorithm for pre-heartbeat Venus versions. In that, because the response depends on publishing `R/<portalid here>/system/0/Connected` first, it's given one cycle extra to get a response. I think maybe we need to learn from the field whether '1 and 2' or '2 and 3' are good values.

Reply:
Makes sense, then let's leave as-is for now, and revisit later if we find a need to do so.